action/candidate_register.go

```diff
@@ -374,15 +380,11 @@ func ethAddrToNativeAddr(in interface{}) (address.Address, error) {
 
 // ToEthTx converts action to eth-compatible tx
 func (cr *CandidateRegister) ToEthTx() (*types.Transaction, error) {
-	addr, err := address.FromString(address.StakingProtocolAddr)
-	if err != nil {
-		return nil, err
-	}
-	ethAddr := common.BytesToAddress(addr.Bytes())
 	data, err := cr.encodeABIBinary()
 	if err != nil {
 		return nil, err
 	}
+	ethAddr := common.BytesToAddress(address.StakingProtocolAddrHash[:])
```

Comment:
easier way of getting the address, compared to 5 lines in the left side

Reply:
also in a new pr

Reply:
ok, will file a new PR